### ISSUE TITLE
Req. for explanation: HA setup

### ISSUE THREAD

**Issue by tokred** [vstakhov/rspamd#2184]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [ ] Other bug
* [x] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [ ] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [x] Not applicable

### Rspamd version:
1.7.2

### Operation system, CPU, memory and environment:
FreeBSD 11.1, redis 3.2

### Description (Please provide a descriptive summary of the issue):
After checking the docs (incl. Using Rspamd in large email systems; Replication in Redis), I am still unsure about redundant email setups and therefore would like to ask for some explanation. In conclusion, is there an approach to run Rspamd in a two node, master-master Postfix/master-master Dovecot setup?

Background: SMTP allows multipe MX records for a domain. Therefore, I can use two Postfix hosts as two distinct MX entries and realise MTA redundancy in an elegant and efficient way -- in case of a host failure, other MTAs will/should automatically try the second MX to deliver emails, so nothing would get lost. Additionally, Dovecot as MDA provides replication of mail storage between hosts via dsync, so users can fetch their emails, e.g. by connecting to an FQDN with two A records, i.e. DNS round-robin.

Now, regarding Rspamd: If those two hosts would run completely isolated/independent Rspamd instances, they a) could both only train on roughly half the emails (=being way less effective), b) would not sync user-classified spam (=being inconsistent for users) and c) would not be able to perform greylisting and proper replies-tagging (=delay emails by double-greylisting).

So, my questions are:
* In general, can one single Redis DB be used by multiple Rspamd instances, incl. concurrent access? If not as a whole, which parts/modules could?
* Unfortunately, Redis only provides master-slave replication. This means if both hosts are alive, Rspamd on the one side would need to stay connected to the master Redis on the other host to perform writes, correct? And if the master dies, the slave stays read-only? What is the behaviour of Rspamd when not being able to reach the write server?
* Is there a better solution to keep both Rspamd (or Redis) instances in sync, e.g. [Netflix/dynomite](https://github.com/Netflix/dynomite)?

Thanks in advance, best regards

**Comment by vstakhov** [comment]
Oh, not this hell again. You never ever need more than master-slave replication in Redis even if you are Google: https://rspamd.com/doc/quickstart.html#using-rspamd-in-large-email-systems

TLDR: if your master is blown up, reconfigure slave as a master and go forward; pauses in learning have negligible impact on spam filtering; autolearning is harmful in the vast majority of the cases
